### ISSUE TITLE
clang broken on ARMv7 with musl

### ISSUE THREAD

**Issue by CMB** [voidlinux/void-packages#5846]
Should've titled this "clang cannot create executables", as clang -c is fine.
Here, hello.c is just the typical hello world program.

$ clang hello.c

/usr/bin/ld: cannot find -lgcc
/usr/bin/ld: cannot find -lgcc
clang-3.9: error: linker command failed with exit code 1 (use -v to see invocation)

$ clang -v hello.c

*snip lots of noise*

 "/usr/bin/ld" -X --eh-frame-hdr -m armelf_linux_eabi -dynamic-linker /lib/ld-musl-armv7lhf.so.1 -o a.out /usr/bin/../lib/crt1.o /usr/bin/../lib/crti.o crtbegin.o -L/usr/bin/../lib -L/lib/../lib -L/usr/lib/../lib -L/usr/bin/../lib -L/lib -L/usr/lib /tmp/hello-2be2e3.o -lgcc --as-needed -lgcc_s --no-as-needed -lc -lgcc --as-needed -lgcc_s --no-as-needed crtend.o /usr/bin/../lib/crtn.o

/usr/bin/ld: cannot find -lgcc

I've also seen it complain about missing crtbegin.o and crtend.o.

**Comment by ebfe** [comment]
still an issue with 4.0.1?

**Comment by CMB** [comment]
Yes, still an issue with 4.0.1.  I've been checking this every time I
see a new clang release, but no joy.

**Comment by chneukirchen** [comment]
Do you have gcc and libgcc-devel installed?

**Comment by CMB** [comment]
Yes I have both, and libgcc-devel also.
